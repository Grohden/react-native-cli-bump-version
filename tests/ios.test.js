const path = require("path");
const { ProjectFilesManager } = require("../lib/index");

const makeDefaultManager = ({
    semver,
    type = "minor",
    skipSemVerFor = "android",
    skipCodeFor = "android",
    pbxFileName = "project.pbxproj",
} = {}) =>
    new ProjectFilesManager({
        type,
        semver,
        skipSemVerFor,
        skipCodeFor,
        root: path.join(__dirname, "ios"),
        pbxprojPath: path.join(__dirname, "ios", pbxFileName),
    });

test("successfully bump version", () => {
    const manager = makeDefaultManager().dryRun();

    expect(manager.pbx.content).toMatchSnapshot();
});

test("skip semVer when asked", () => {
    const manager = makeDefaultManager({ skipSemVerFor: "all" }).dryRun();

    expect(manager.pbx.content).toMatchSnapshot();
});

test("direct set semver string", () => {
    const manager = makeDefaultManager({ semver: "1.1.2" }).dryRun();

    expect(manager.pbx.content).toMatchSnapshot();
    expect(manager.packageJSON.content).toMatchSnapshot();
});
