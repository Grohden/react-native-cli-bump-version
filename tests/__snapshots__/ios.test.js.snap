// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`direct set semver string 1`] = `
"// !$*UTF8*$!
{
	objects = {
/* Begin XCBuildConfiguration section */
		13B07F941A680F5B01A75B9A /* Debug */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				MARKETING_VERSION = 1.1.2;
				CURRENT_PROJECT_VERSION = 32;
			};
			name = Debug;
		};
		13B07F951A680F5B01A75B9A /* Release */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				MARKETING_VERSION = 1.1.2;
				CURRENT_PROJECT_VERSION = 32;
			};
			name = Release;
		};
/* End XCBuildConfiguration section */
	};
}
"
`;

exports[`direct set semver string 2`] = `
Object {
  "version": "1.1.2",
}
`;

exports[`skip semVer when asked 1`] = `
"// !$*UTF8*$!
{
	objects = {
/* Begin XCBuildConfiguration section */
		13B07F941A680F5B01A75B9A /* Debug */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				MARKETING_VERSION = 1.0.0;
				CURRENT_PROJECT_VERSION = 32;
			};
			name = Debug;
		};
		13B07F951A680F5B01A75B9A /* Release */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				MARKETING_VERSION = 1.0.0;
				CURRENT_PROJECT_VERSION = 32;
			};
			name = Release;
		};
/* End XCBuildConfiguration section */
	};
}
"
`;

exports[`successfully bump version (pre 0.69.x) 1`] = `
"// !$*UTF8*$!
{
	objects = {
/* Begin XCBuildConfiguration section */
		13B07F941A680F5B01A75B9A /* Debug */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				MARKETING_VERSION = 1.1.0;
				CURRENT_PROJECT_VERSION = 32;
			};
			name = Debug;
		};
		13B07F951A680F5B01A75B9A /* Release */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				MARKETING_VERSION = 1.1.0;
				CURRENT_PROJECT_VERSION = 32;
			};
			name = Release;
		};
/* End XCBuildConfiguration section */
	};
}
"
`;

exports[`successfully bump version 1`] = `
"// !$*UTF8*$!
{
	objects = {
/* Begin XCBuildConfiguration section */
		13B07F941A680F5B01A75B9A /* Debug */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				MARKETING_VERSION = 1.1.0;
				CURRENT_PROJECT_VERSION = 32;
			};
			name = Debug;
		};
		13B07F951A680F5B01A75B9A /* Release */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				MARKETING_VERSION = 1.1.0;
				CURRENT_PROJECT_VERSION = 32;
			};
			name = Release;
		};
/* End XCBuildConfiguration section */
	};
}
"
`;
